audeer/core/utils.py

```diff
@@ -488,15 +488,34 @@ def is_uid(uid: str) -> bool:
     Returns:
         ``True`` if string is a unique identifier
 
+    Examples:
+        >>> is_uid(uid())
+        True
+        >>> is_uid(uid(short=True))
+        True
+        >>> is_uid('not a unique identifier')
+        False
+
     """
     if uid is None:
         return False
     if not isinstance(uid, str):
         return False
+    if len(uid) != 8 and len(uid) != 36:
+        return False
+
+    if len(uid) == 8:
+        uid = f'00000000-0000-0000-0000-0000{uid}'
+
+    for pos in [8, 13, 18, 23]:
+        if not uid[pos] == '-':
+            return False
```

Comment:
Does this mean we had a bug before and were returning `True` for UIDs with `-` at the wrong position?

Reply:
Yes, it was returning `True` when I added a test where the `-` was in a wrong position.